server/src/main/java/org/jboss/as/server/controller/resources/ModuleInfoHandler.java

```diff
@@ -0,0 +1,202 @@
+            SimpleAttributeDefinitionBuilder.create("export-filter", ModelType.STRING).build(),
+            SimpleAttributeDefinitionBuilder.create("import-filter", ModelType.STRING).build(),
+            SimpleAttributeDefinitionBuilder.create("optional", ModelType.STRING).build(),
+            SimpleAttributeDefinitionBuilder.create("local-loader-class", ModelType.STRING).build(),
+            new StringListAttributeDefinition.Builder("local-loader-paths").build()
+    )
+            .setStorageRuntime()
+            .build();
+
+    static final AttributeDefinition DEPENDENCIES = ObjectListAttributeDefinition.Builder.of("dependencies", DEPENDENCY)
+            .setStorageRuntime()
+            .build();
+
+    static final ObjectTypeAttributeDefinition RESOURCE_LOADER = ObjectTypeAttributeDefinition.Builder.of("resource-loader",
+            SimpleAttributeDefinitionBuilder.create("type", ModelType.STRING).build(),
+            new StringListAttributeDefinition.Builder("paths").build()
+
+    )
+            .setStorageRuntime()
+            .build();
+
+
+    static final AttributeDefinition RESOURCE_LOADERS = ObjectListAttributeDefinition.Builder.of("resource-loaders", RESOURCE_LOADER)
+            .setStorageRuntime()
+            .build();
+
+
+    public static final OperationDefinition DEFINITION = new SimpleOperationDefinitionBuilder("module-info", ModuleLoadingResourceDefinition.INSTANCE.getResourceDescriptionResolver())
+            .setParameters(MODULE_NAME, MODULE_SLOT)
+            .setRuntimeOnly()
+            .setReadOnly()
+            .setReplyType(ModelType.OBJECT)
+            .setReplyParameters(MODULE_NAME, MAIN_CLASS, FALLBACK_LOADER, DEPENDENCIES, RESOURCE_LOADERS)
+            .build();
+
+    public static ModuleInfoHandler INSTANCE = new ModuleInfoHandler();
+
+    private ModuleInfoHandler() {
+        super(DEFINITION);
+    }
+
+
+    @Override
+    public void performRuntime(final OperationContext context, final ModelNode operation, final ModelNode model) throws OperationFailedException {
+        String moduleName = MODULE_NAME.resolveModelAttribute(context, model).asString();
+        String slot = MODULE_SLOT.resolveModelAttribute(context, model).asString();
+        ModuleIdentifier id = ModuleIdentifier.create(moduleName, slot);
+        ModuleLoader loader = Module.getBootModuleLoader();
+        try {
+            ModuleLoaderMXBean mxBean = getMxBean(loader);
+            ModuleInfo moduleInfo = mxBean.getModuleDescription(id.toString());
+            context.getResult().set(displayModuleInfo(moduleInfo));
+        } catch (Exception e) {
+            throw new OperationFailedException("Could not get module info", e);
+        }
+    }
+
+
+    private ModuleLoaderMXBean getMxBean(ModuleLoader loader) throws ReflectiveOperationException {
+        Field mxBeanField = ModuleLoader.class.getDeclaredField("mxBean");
```

Comment:
In today's Domain Management call David said he's fine with the reflection approach.